tls/s2n_psk.c

```diff
@@ -291,9 +317,30 @@ int s2n_offered_psk_list_choose_psk(struct s2n_offered_psk_list *psk_list, struc
         psk_params->chosen_psk = NULL;
         return S2N_SUCCESS;
     }
-
-    POSIX_GUARD_RESULT(s2n_match_psk_identity(&psk_params->psk_list, &psk->identity, &psk_params->chosen_psk));
-    psk_params->chosen_psk_wire_index = psk->wire_index;
+    
+    if(psk_params->type == S2N_PSK_TYPE_EXTERNAL) {
+        POSIX_GUARD_RESULT(s2n_match_psk_identity(&psk_params->psk_list, &psk->identity, &psk_params->chosen_psk));
+        psk_params->chosen_psk_wire_index = psk->wire_index;
+    } else {
+        struct s2n_stuffer ticket_stuffer = { 0 };
```

Comment:
Why did we get rid of the `s2n_match_external_psk` and `s2n_match_resumption_psk` helper functions, I liked the separation and it made it easier to read?

Reply:
We wanted to cover the case where the customer has a callback set to choose the psk, but the mode is resumption. The customer could select a psk using s2n_offered_psk_list_choose_psk() and then we'd be able to decrypt the identity and resume a session if the decrypted info was valid. It's not a use-case that we expect people will want, but disabling it is more work than making it work.